process/block/preprocess/transactions.go

```diff
@@ -1351,13 +1351,12 @@ func (txs *transactions) ProcessMiniBlock(
 
 	defer func() {
 		if err != nil {
-			if scheduledMode {
-				txs.gasHandler.RemoveGasProvidedAsScheduled(processedTxHashes)
-			} else {
-				txs.gasHandler.RemoveGasProvided(processedTxHashes)
-				txs.gasHandler.RemoveGasRefunded(processedTxHashes)
-				txs.gasHandler.RemoveGasPenalized(processedTxHashes)
+			for _, hash := range processedTxHashes {
+				//TODO: Remove it or set log level to Trace
+				log.Debug("transactions.ProcessMiniBlock: defer func()", "tx hash", hash)
 			}
+
+			txs.gasHandler.RestoreGasSinceLastReset()
```

Comment:
the logic before for scheduledMode called only removedTheGasProvidedAsScheduled. Now there will be a reset for all gascomputations.  Is it ok?

Reply:
The mechanism has changed and the Reset method called before starting to process, resets all four lists of tx hashes (gas provided, gasProvidedAsScheduled, gasRefunded and gasPenalized). Then when processing is started the Set method for (gasProvided, gasProvidedAsScheduled, gasRefunded and gasPenalized) is called accordingly with the current mode (scheduled or not). So the RestoreGasSinceLastReset will have nothing to restore for gasProvidedAsScheduled when the scheduled mode is false an vice-versa will have nothing to restore for gasProvided, gasRefunded and gasPenalized when the scheduled mode is true. In case of scheduled mode true the transaction is not really executed, just the basic checks are done and so the only thing which could be calculated is gasprovidedAsScheduled. Afterwards the scheduled txs are executed and resulted scrs, accumulated fees, developer fees, etc, are added in the next block.

Reply:
ok!